src/librustc_mir/util/aggregate.rs

```diff
@@ -53,13 +53,12 @@ pub fn expand_aggregate<'tcx>(
         .map(move |(i, (op, ty))| {
             let lhs_field = if let AggregateKind::Array(_) = kind {
                 // FIXME(eddyb) `offset` should be u64.
-                let offset = i as u32;
+                let offset = i as u64;
                 assert_eq!(offset as usize, i);
```

Comment:
AFAIK this dance isn't needed anymore, since `usize` is at most `u64` currently, but cc @RalfJung who removed a bunch of `as` casts from miri recently.

Reply:
I'd use `u64::try_from(...).unwrap()` instead of a manual `assert_eq!`.

I changed `Size::from_bytes` so that it does the (checked) cast itself, but here it looks like we need it for `ConstantIndex` and I didn't touch that.